test/persistence-hooks.suite.js

```diff
@@ -1095,6 +1095,31 @@ module.exports = function(dataSource, should) {
         });
       });
 
+      it('applies propagates hookState from `before delete` to `after delete` hook', function(done) {
```

Comment:
Applies or propagates? Pick only one please.

Also please add a check that the hookState is propagated from the "access" hook too.